internal/formatters/fmt_base.go

```diff
@@ -21,12 +21,12 @@ import (
 )
 
 // BaseFormatterFunc implements the FormatterFunc for the base formatter.
-func BaseFormatterFunc(suite string, out io.Writer) formatters.Formatter {
+func BaseFormatterFunc(suite string, out io.WriteCloser) formatters.Formatter {
 	return NewBase(suite, out)
 }
 
 // NewBase creates a new base formatter.
-func NewBase(suite string, out io.Writer) *Base {
+func NewBase(suite string, out io.WriteCloser) *Base {
 	return &Base{
 		suiteName: suite,
 		indent:    2,
```

Comment:
Regarding this approach, do you think it will force the users (who use custom fmt) to working too much on their end to adapt to our changes? I think having a `Close()` method is a great enhancement, but I'm considering the case that someone is having problem with hook ordering bug, and expecting the new release will fix it, but then in order to use it they need to update their custom code first...

Should we use another way such as saying "we're going to fix this, and also we're migrating the Formatter API to a new version, you don't need to update anything yet and still have new features to use"? I have the following code that seems to work on my local testing
```suggestion
type baseAdapter struct {
	out io.Writer
}

func (f baseAdapter) Write(p []byte) (n int, err error) {
	return f.out.Write(p)
}

func (f baseAdapter) Close() error {
	return nil
}

// NewBase creates a new base formatter.
func NewBase(suite string, out io.Writer) *Base {
	var wc io.WriteCloser

	w, ok := out.(io.WriteCloser)
	if ok {
		wc = w
	} else {
		wc = baseAdapter{out: out}
	}

	return &Base{
		suiteName: suite,
		indent:    2,
		out:       wc,
		Lock:      new(sync.Mutex),
	}
}

// Base is a base formatter.
type Base struct {
	suiteName string
	out       io.WriteCloser
	indent    int

	Storage *storage.Storage
	Lock    *sync.Mutex
}
```

This way we can still migrate to use `io.WriteCloser` in our internal logic, but user don't need to know or don't need to update anything right away.
Ps: sorry my english is not that good so my reasoning is kinda long.